pkg/operators/common/systemdcache.go

```diff
@@ -0,0 +1,219 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/coreos/go-systemd/v22/dbus"
+	"github.com/sirupsen/logrus"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/container-utils/cgroups"
+)
+
+type SystemdSubInterface interface {
+	AddCgroup(string, uint64)
+	RemoveCgroup(string, uint64)
+}
+
+type SystemdCache struct {
```

Comment:
IIUC, this `SystemdCache` is a unique cache that keeps an up-to-date list of systemd units and is used by two operators for different purposes:

1. The systemd resolver: Enrich the events with the systemd unit names. The cache returns the systemd unit name from the CgroupID. 
2. The local manager: Keep the `CgroupMap` up-to-date with the existing systemd units. The `CgroupMap` is a map BPF programs use to filter by cgroup when `--sdunit` is passed.

Is my understanding correct?

For the `--sdunit` flag, we need to clarify that the very first events of the requested unit might not be captured as the notification of the new systemd unit is not synchronous as the container-hook does for the containers.

Reply:
Is now synchronous. Thanks to you and @alban :)